odoo/tools/misc.py

```diff
@@ -1656,38 +1656,51 @@ def __getattr__(self, attrib):
         return DotDict(val) if type(val) is dict else val
 
 
-def get_diff(data_from, data_to, custom_style=False):
+def get_diff(data_from, data_to, custom_style=False, dark_color_scheme=False):
     """
     Return, in an HTML table, the diff between two texts.
 
     :param tuple data_from: tuple(text, name), name will be used as table header
     :param tuple data_to: tuple(text, name), name will be used as table header
     :param tuple custom_style: string, style css including <style> tag.
+    :param tuple dark_color_scheme: boolean, true if dark color scheme is used
     :return: a string containing the diff in an HTML table format.
     """
-    def handle_style(html_diff, custom_style):
+    def handle_style(html_diff, custom_style, dark_color_scheme):
         """ The HtmlDiff lib will add some useful classes on the DOM to
         identify elements. Simply append to those classes some BS4 ones.
         For the table to fit the modal width, some custom style is needed.
         """
         to_append = {
             'diff_header': 'bg-600 text-center align-top px-2',
             'diff_next': 'd-none',
-            'diff_add': 'bg-success',
-            'diff_chg': 'bg-warning',
-            'diff_sub': 'bg-danger',
         }
         for old, new in to_append.items():
             html_diff = html_diff.replace(old, "%s %s" % (old, new))
         html_diff = html_diff.replace('nowrap', '')
+        colors = ('#7f2d2f', '#406a2d', '#51232f', '#3f483b') if dark_color_scheme else (
```

Comment:
Is there actually any reason to retrieve the dark mode value in the caller and pass it through methods instead of doing it here?
Eg something like
```suggestion
        colors = ('#7f2d2f', '#406a2d', '#51232f', '#3f483b') if request and request.httprequest.cookies.get('color_scheme') == 'dark' else (
```

Reply:
Because `request` is not imported, and when I try to do it in that file I get this traceback:
```py
Traceback (most recent call last):
  File "/usr/local/bin/odoo", line 5, in <module>
    import odoo
  File "/home/odoo/src/odoo/odoo/__init__.py", line 113, in <module>
    from . import modules
  File "/home/odoo/src/odoo/odoo/modules/__init__.py", line 8, in <module>
    from . import db, graph, loading, migration, module, registry, neutralize
  File "/home/odoo/src/odoo/odoo/modules/graph.py", line 10, in <module>
    import odoo.tools as tools
  File "/home/odoo/src/odoo/odoo/tools/__init__.py", line 7, in <module>
    from . import pdf
  File "/home/odoo/src/odoo/odoo/tools/pdf.py", line 43, in <module>
    from odoo.tools.misc import file_open
  File "/home/odoo/src/odoo/odoo/tools/misc.py", line 46, in <module>
    from odoo.http import request
  File "/home/odoo/src/odoo/odoo/http.py", line 164, in <module>
    from .modules.module import get_manifest
  File "/home/odoo/src/odoo/odoo/modules/module.py", line 65, in <module>
    @tools.lazy
AttributeError: partially initialized module 'odoo.tools' has no attribute 'lazy' (most likely due to a circular import)
```

Reply:
Whoops, indeed, it's in `tools`, odoo stuff should be avoided there, it's not supposed to rely on Odoo